### DISCUSSION TITLE
Delete application files on uninstall

### DISCUSSION THREAD

**Discussion by TristanPouliquen** [tauri-apps/tauri#6497] [Q&A] [answered]
Hello, 

My app creates some files in the `%APPDATA%` directory. When a user uninstalls my app, I would like these files to be deleted, so as to not leave useless files on my users' computer.

As is, this does not seem to work, at least on Windows.

Is there a way to make this work?

Deleting app data seems a standard procedure for me.

**Comment by simonhyll** [comment, marked as answer]
My best guess is that the WIX installer isn't configured to do that. This however can be fairly easily taken care of with the new NSIS based installer. I haven't tried it myself yet so can't tell you exactly how, just offer the comfort that there is a way of doing it 😅

**Reply by TristanPouliquen** [reply]
Hello, do you have somewhere I can see how to use this NSIS installer? 

In https://tauri.app/v1/guides/building/windows/, it talks only of Wix and does not give insight into how to switch to another.

**Reply by FabianLars** [reply]
The guide will be available once tauri 1.3 will be released (the nsis installer is part of tauri 1.3)

This should be fairlyyy straight forward to add by modifying the wix template on your end (well, as straight forward as wix and xml can be lol).
nsis will indeed have a built-in option for this, implemented here https://github.com/tauri-apps/tauri/pull/6129.
In theory it should be enough to consume [tauri from git](https://tauri.app/v1/guides/faq#how-can-i-use-unpublished-tauri-changes), and if you have it configured to build all bundle targets, a normal `tauri build` shoud get you an nsis bundle too.

**Reply by TristanPouliquen** [reply]
Thanks, yes I was looking into Wix fragments, with the `RemoveFolderEx` directive in particular.

I was looking at the  base Wix file from Tauri (https://github.com/tauri-apps/tauri/blob/2f70d8da2bc079400bb49e6793f755306049aab2/tooling/bundler/src/bundle/windows/templates/main.wxs), and found this guide (https://www.hass.de/content/wix-how-use-removefolderex-your-xml-scripts) to guide me through adding the necessary directives. 

I will update you on if it works :-)

Thanks again for your time!

**Reply by TristanPouliquen** [reply]
Do you have an idea as to when Tauri v1.3 will be going out? :) 

Will you completely drop the MSI installer? Or could it be useful if I make a PR adding the deletion of these files to the base template?

**Reply by FabianLars** [reply]
> Do you have an idea as to when Tauri v1.3 will be going out? :)

Soon-ish. I don't know if we have an internal ETA but we're not pushing any more features to dev because we want to get it out, it was already for quite some time in internal audit.

> Will you completely drop the MSI installer?

No, it still has a valid use-case in corporate environments for example (or for some "normal" target audiences), our focus will just swap to NSIS as our main bundle in v2.

> Or could it be useful if I make a PR adding the deletion of these files to the base template?

I personally don't see anything wrong with adding it, at least if you can match the NSIS implementation of being a checkbox in the uninstaller. Also, make sure to target the `next` branch because of what i said above about `dev` being frozen because of 1.3

**Reply by TristanPouliquen** [reply]
Yeah, I tried digging a bit into MSI fragments etc, did not find an easy solution and my use-case is a oneshot, where my users will install & version to test out, and reinstall at a later date when I'll have additional data.

I'll just add a diesel migration and let it be run automatically I think, even if it' less clean, but this is for a one-shot in 3 weeks time, and will probably not last much longer so I don't want to spend too much time on this 😅 

Thank you again to both of you :)

**Reply by TristanPouliquen** [reply]
And from what I quickly read, uninstalling a .msi does not alwasy show a UI, so matching the NSIS implementation might be tricky!
